iree/tools/run_lit.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import re
+import subprocess
+from typing import List, Tuple, Union
+
+
+def parse_arguments():
+  parser = argparse.ArgumentParser(description="run_lit.py helper.")
+  parser.add_argument("--test_file", help="Path of the test file to execute.",
+                      required=True)
+  args = parser.parse_args()
+  return args
+
+
+def execute_in_shell(command: str,
+                     shell: str = '/bin/bash') -> Tuple[Union[str, int]]:
+  """Executes command and returns its stdout, stderr and returncode."""
+  p = subprocess.Popen([shell, command], shell=True, stdout=subprocess.PIPE,
```

Comment:
What happens if there's a pipefail in this environment? In bash if one of the commands in a pipe fails, it happily continues. We `set -o pipefail` in run_lit.sh to error out instead. Can we do that here too?

```shell
$ blarg | echo foo
foo
-bash: blarg: command not found
$ echo $?
0
```

In a pinch we can insert it in the bash command:

```shell
$ set -o pipefail; blarg | echo foo
foo
-bash: blarg: command not found
$ echo $?
127
```